packages/mattermost-redux/src/reducers/entities/threads/threadsInTeam.ts

```diff
@@ -45,8 +45,20 @@ function handlePostRemoved(state: State, action: GenericAction) {
     };
 }
 
-export function handleReceivedThread(state: State, action: GenericAction, extra: ExtraData) {
-    const {thread, team_id: teamId} = action.data;
+// adds thread to all teams in state
+function handleAllTeamsReceivedThread(state: State, thread: UserThread, teamId: Team['id'], extra: ExtraData) {
+    const teamIds = Object.keys(state);
+
+    let newState = {...state};
+    for (const teamId of teamIds) {
+        newState = handleSingleTeamReceivedThread(newState, thread, teamId, extra);
```

Comment:
If we are going this route, shouldn't we also add these threads to a new team if/when we join a new team? (therefore applying the changes in the websocket events and around the client calls to join a team)

Reply:
if the store holds no threads in a team we cannot add any single thread in the global threads, 
the reason is that we could break pagination, old.
When you first visit global threads in a team we call for getUserThreads, and then we can deduce whether to a add a single thread or not. 
The criteria is that we can add threads newer than the last thread we have in store, but not threads older than that one.

Reply:
older threads will get added via scrolling/paginating

Reply:
Let me know what happens in the following scenario:
- I have 3 threads on DM/GMs with mentions
- I join team B
- Threads show:
  - Option A: Nothing, since threads are not yet loaded
  - Option B: A badge with the number 3, corresponding to the 3 DM/GMs unread

From what I understand from what you explained, what happens is Option A, and I would consider that a bug. Even if we try to solve it with this approach, it would seem really error prone. Having DMs/GMs where they belong (in no team) instead of repeated on every team seems more reliable.

Another thing to consider is what happens if a thread is removed? (Root post is deleted, or anything of that sort)

Reply:
@larkox Option B will happen. Counts store is separate from threads store. 
Imagine just refreshing the app, you have no threads in store, but we still need show the mention count.